### ISSUE TITLE
[BUG] multielasticdump only dumps 6 indices

### ISSUE THREAD

**Issue by willfurnell** [taskrabbit/elasticsearch-dump#627]
**Context**
- ElasticDump version: 6.14.1
- Elasticsearch version: 5.6.16
- Node.js version *(note that nodejs v8+ is required for this tool)*: 12.11.1
- Full command you are having issue with (with inputs and outputs)

```multielasticdump --input=http://localhost:9200/ --output=/media/dump/ --limit=10000 --noRefresh```

**Describe the bug**
Multielasticdump only gets the first six indices when dumping - I am trying to dump 180ish indices.

**To Reproduce**
Run the command when you have more than 6 indices
**Current behavior**
6 indices are dumped

**Expected behavior**
All indices on the server are dumped

This dump is being performed on RHEL 7.6

**Comment by ferronrsmith** [comment]
> A good bug report shouldn't leave others needing to chase you up for more information. Please try to be as detailed as possible in your report. 


There's absolutely nothing here to replicate the issue. I tested locally and remotely and was unable to replicate the issue.

**Comment by willfurnell** [comment]
What would you like me to provide so you can try and replicate the issue?

**Comment by willfurnell** [comment]
This is the output of the aliases endpoint that I can see elasticdump calls: `{"graylog_1862":{"aliases":{}},"graylog_1987":{"aliases":{}},"graylog_1864":{"aliases":{}},"graylog_1972":{"aliases":{}},"graylog_1990":{"aliases":{}},"graylog_1876":{"aliases":{}},"graylog_1840":{"aliases":{}},"graylog_1841":{"aliases":{}},"graylog_1885":{"aliases":{}},"graylog_1938":{"aliases":{}},"graylog_1899":{"aliases":{}},".monitoring-es-6-2019.10.15":{"aliases":{}},"graylog_1893":{"aliases":{}},"graylog_1823":{"aliases":{}},"graylog_1855":{"aliases":{}},"graylog_1833":{"aliases":{}},".watches":{"aliases":{}},"graylog_1984":{"aliases":{}},"graylog_1949":{"aliases":{}},"graylog_1935":{"aliases":{}},"graylog_1831":{"aliases":{}},"graylog_1887":{"aliases":{}},"graylog_1923":{"aliases":{}},".monitoring-kibana-6-2019.10.15":{"aliases":{}},"graylog_1889":{"aliases":{}},"graylog_1867":{"aliases":{}},"graylog_1859":{"aliases":{}},"graylog_1868":{"aliases":{}},"graylog_1909":{"aliases":{}},"graylog_1939":{"aliases":{}},"graylog_1962":{"aliases":{}},"graylog_1895":{"aliases":{}},"graylog_1861":{"aliases":{}},"graylog_1936":{"aliases":{}},"graylog_1911":{"aliases":{}},"graylog_1958":{"aliases":{}},"graylog_1921":{"aliases":{}},"graylog_1907":{"aliases":{}},"graylog_1940":{"aliases":{}},"graylog_1963":{"aliases":{}},"graylog_1842":{"aliases":{}},"graylog_1818":{"aliases":{}},"graylog_1913":{"aliases":{}},"graylog_1976":{"aliases":{}},"graylog_1830":{"aliases":{}},"graylog_1912":{"aliases":{}},"graylog_1942":{"aliases":{}},"graylog_1866":{"aliases":{}},"graylog_1871":{"aliases":{}},"graylog_1943":{"aliases":{}},"graylog_1835":{"aliases":{}},"graylog_1854":{"aliases":{}},"graylog_1988":{"aliases":{}},"graylog_1941":{"aliases":{}},"graylog_1882":{"aliases":{}},"graylog_1978":{"aliases":{}},"graylog_1982":{"aliases":{}},".monitoring-es-6-2019.10.16":{"aliases":{}},"graylog_1886":{"aliases":{}},"graylog_1970":{"aliases":{}},"graylog_1901":{"aliases":{}},"graylog_1888":{"aliases":{}},".watcher-history-6-2019.10.16":{"aliases":{}},"graylog_1843":{"aliases":{}},"graylog_1973":{"aliases":{}},"graylog_1837":{"aliases":{}},"graylog_1930":{"aliases":{}},"graylog_1820":{"aliases":{}},"graylog_1910":{"aliases":{}},"graylog_1877":{"aliases":{}},"graylog_1881":{"aliases":{}},"graylog_1926":{"aliases":{}},".watcher-history-6-2019.10.15":{"aliases":{}},"graylog_1872":{"aliases":{}},"graylog_1898":{"aliases":{}},"graylog_1917":{"aliases":{}},"graylog_1834":{"aliases":{}},"graylog_1914":{"aliases":{}},"graylog_1874":{"aliases":{}},"graylog_1985":{"aliases":{}},"graylog_1836":{"aliases":{}},"graylog_1925":{"aliases":{}},"graylog_1849":{"aliases":{}},"graylog_1974":{"aliases":{}},"graylog_1933":{"aliases":{}},"graylog_1934":{"aliases":{}},"graylog_1880":{"aliases":{}},"graylog_1944":{"aliases":{}},"graylog_1980":{"aliases":{}},"graylog_1897":{"aliases":{}},"graylog_1822":{"aliases":{}},"graylog_1994":{"aliases":{"graylog_deflector":{}}},"graylog_1860":{"aliases":{}},"graylog_1838":{"aliases":{}},"graylog_1981":{"aliases":{}},"graylog_1863":{"aliases":{}},"graylog_1850":{"aliases":{}},"graylog_1979":{"aliases":{}},"graylog_1879":{"aliases":{}},"graylog_1956":{"aliases":{}},"graylog_1894":{"aliases":{}},".monitoring-alerts-6":{"aliases":{}},"graylog_1955":{"aliases":{}},"graylog_1977":{"aliases":{}},"graylog_1950":{"aliases":{}},"graylog_1821":{"aliases":{}},"graylog_1960":{"aliases":{}},"graylog_1915":{"aliases":{}},"graylog_1961":{"aliases":{}},"graylog_1826":{"aliases":{}},"graylog_1890":{"aliases":{}},"graylog_1892":{"aliases":{}},"graylog_1875":{"aliases":{}},"graylog_1929":{"aliases":{}},"graylog_1891":{"aliases":{}},"graylog_1975":{"aliases":{}},"graylog_1873":{"aliases":{}},".triggered_watches":{"aliases":{}},"graylog_1953":{"aliases":{}},"graylog_1908":{"aliases":{}},"graylog_1959":{"aliases":{}},"graylog_1847":{"aliases":{}},".kibana":{"aliases":{}},"graylog_1903":{"aliases":{}},"graylog_1851":{"aliases":{}},"graylog_1937":{"aliases":{}},"graylog_1825":{"aliases":{}},"graylog_1846":{"aliases":{}},"graylog_1947":{"aliases":{}},"graylog_1845":{"aliases":{}},"graylog_1966":{"aliases":{}},"graylog_1969":{"aliases":{}},"graylog_1832":{"aliases":{}},"graylog_1848":{"aliases":{}},"graylog_1819":{"aliases":{}},"graylog_1869":{"aliases":{}},"graylog_1983":{"aliases":{}},"graylog_1945":{"aliases":{}},"graylog_1952":{"aliases":{}},".monitoring-kibana-6-2019.10.16":{"aliases":{}},"graylog_1971":{"aliases":{}},"graylog_1928":{"aliases":{}},"graylog_1922":{"aliases":{}},"graylog_1828":{"aliases":{}},"graylog_1919":{"aliases":{}},"graylog_1992":{"aliases":{}},"graylog_1951":{"aliases":{}},"graylog_1865":{"aliases":{}},"graylog_1816":{"aliases":{}},"graylog_1993":{"aliases":{}},"graylog_1857":{"aliases":{}},"graylog_1815":{"aliases":{}},"graylog_1906":{"aliases":{}},"graylog_1844":{"aliases":{}},"graylog_1904":{"aliases":{}},"graylog_1905":{"aliases":{}},"graylog_1920":{"aliases":{}},"graylog_1967":{"aliases":{}},"graylog_1878":{"aliases":{}},"graylog_1829":{"aliases":{}},"graylog_1957":{"aliases":{}},"graylog_1852":{"aliases":{}},"graylog_1839":{"aliases":{}},"graylog_1946":{"aliases":{}},"graylog_1902":{"aliases":{}},"graylog_1817":{"aliases":{}},"graylog_1883":{"aliases":{}},"graylog_1858":{"aliases":{}},"graylog_1927":{"aliases":{}},"graylog_1924":{"aliases":{}},"graylog_1989":{"aliases":{}},"graylog_1965":{"aliases":{}},"graylog_1870":{"aliases":{}},"graylog_1884":{"aliases":{}},"graylog_1964":{"aliases":{}},"graylog_1986":{"aliases":{}},"graylog_1932":{"aliases":{}},"graylog_1991":{"aliases":{}},"graylog_1931":{"aliases":{}},"graylog_1900":{"aliases":{}},"graylog_1856":{"aliases":{}},"graylog_1948":{"aliases":{}},"graylog_1918":{"aliases":{}},"graylog_1853":{"aliases":{}},"graylog_1916":{"aliases":{}},"graylog_1968":{"aliases":{}},"graylog_1827":{"aliases":{}},"graylog_1954":{"aliases":{}},"graylog_1824":{"aliases":{}},"graylog_1896":{"aliases":{}}}`

**Comment by ferronrsmith** [comment]
1. Logs (terminal output)
1. ES list of indices would help (eve though the default regex is `^.*$`, which matches everything.
1. _alias output

**Comment by willfurnell** [comment]
I'm trying a dump of the data only now. Probably also worth noting (only just noticed this myself) that the machine has 6 cores.

**Comment by ferronrsmith** [comment]
6 just happens to be how many jobs are ran in parallel. Can you try running the docker container, because I can't seem to replicate the issue, which means it might be a system issue.

**Comment by willfurnell** [comment]
Interestingly I still have the same issue in the docker container (which I ran on a Centos 7 rather than RHEL 7 VM, but with the same specs) - 7 indices were downloaded.

**Comment by willfurnell** [comment]
Sure, here's the output from running in the docker container

**Comment by shawnz** [comment]
I am also having an issue where multielasticdump does not complete all indexes and prints the "child process exited with error" at some point during the output. Was the change made to print the errors already? If so how do I enable that functionality?

**Comment by ferronrsmith** [comment]
Yes send the logs

**Comment by shawnz** [comment]
Here is an example of a run with the problem:

https://gist.github.com/shawnz/ea58d60979b08ce3d7a0b9b2a13a7a8b

Is it an out of memory issue? Would it help to reduce the number of threads? I'd prefer not to lower the "limit" since that seems to decrease network performance greatly.

**Comment by shawnz** [comment]
Upgrading from node 8 to node 12 appears to have resolved the issue.

Thank you for the help and for this very useful application.

**Comment by ferronrsmith** [comment]
Very long string means you're dumping too much data. The string is too large to be parsed by node.

**Comment by shawnz** [comment]
With limit 1000 I got a different error, ENOMEM: https://gist.github.com/shawnz/7385a97bc3b36f52c37c802220ce907c

With limit 10000 parallelism 2 I got a successful result finally: https://gist.github.com/shawnz/22fc8a144aa6ae50796000354160bbfc

The network utilization was not as good as with parallelism 12, but now that I have my data backed up I can spend some more time to worry about the best set of options.

Thanks for your help and for your work on this excellent product. It will be very helpful for us.

If you're interested, my system configuration is: 12 HW threads, 16 GB RAM, nodejs 12 running on windows/WSL, and I am backing up to a spinning metal HDD.
